WireAPI/Tests/WireAPITests/UsersAPITests.swift

```diff
@@ -0,0 +1,176 @@
+
+    // MARK: - V0
+
+    func testGetUserForID_SuccessResponse_200_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetUserSuccessResponseV0"
+        )
+
+        let sut = UsersAPIV0(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getUser(for: Scaffolding.userID)
+
+        // Then
+        XCTAssertEqual(
+            result,
+            Scaffolding.user
+
+        )
+    }
+
+    func testGetUsersForIDs_SuccessResponse_200_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetUsersSuccessResponseV0"
+        )
+        let sut = UsersAPIV0(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getUsers(userIDs: [Scaffolding.userID])
+
+        // Then
+        XCTAssertEqual(
+            result,
+            ListUsersResponse(found: [Scaffolding.user], failed: [])
+        )
+    }
+
+    func testGetUsersForIDs_FailureResponse_NotFound_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 404, errorLabel: "not-found")
+        let sut = UsersAPIV0(httpClient: httpClient)
+
+        // Then
+        await assertAPIError(UsersAPIError.notFound) {
+            // When
+            _ = try await sut.getUser(for: Scaffolding.userID)
+        }
+    }
+
+    // MARK: - V4
+
+    func testGetUserForID_SuccessResponse_200_V4() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetUserSuccessResponseV4"
+        )
+
+        let sut = UsersAPIV4(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getUser(
+            for: Scaffolding.userID
+        )
+
+        // Then
+        XCTAssertEqual(
+            result,
+            Scaffolding.user
+
+        )
+    }
+
+    func testGetUsersForIDs_FailureResponse_NotFound_V4() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 404, errorLabel: "not-found")
+        let sut = UsersAPIV4(httpClient: httpClient)
+
+        // Then
+        await assertAPIError(UsersAPIError.notFound) {
+            // When
+            _ = try await sut.getUser(for: Scaffolding.userID)
+        }
+    }
+
+    func testGetUsersForIDs_SuccessResponse_200_V4() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetUsersSuccessResponseV4"
+        )
+        let sut = UsersAPIV4(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getUsers(userIDs: [Scaffolding.userID])
+
+        // Then
+        XCTAssertEqual(
+            result,
+            ListUsersResponse(found: [Scaffolding.user], failed: [Scaffolding.userID])
+        )
+    }
+
+    struct Scaffolding {
```

Comment:
Is scaffolding just a container of test data?

Reply:
yes

Reply:
Happy to choose a better name

Reply:
nah it's cool was just wondering.